synapse/storage/data_stores/main/cache.py

```diff
@@ -73,17 +82,24 @@ def _invalidate_state_caches_and_stream(self, txn, room_id, members_changed):
                 txn, CURRENT_STATE_CACHE_NAME, [room_id]
             )
 
-    def _send_invalidation_to_replication(self, txn, cache_name, keys):
+    def _send_invalidation_to_replication(
+        self, txn, cache_name: str, keys: Optional[Iterable[Any]]
+    ):
         """Notifies replication that given cache has been invalidated.
 
         Note that this does *not* invalidate the cache locally.
 
         Args:
             txn
-            cache_name (str)
-            keys (iterable[str])
+            cache_name
+            keys: Entry to invalidate. If None will invalidate all.
         """
 
+        if cache_name == CURRENT_STATE_CACHE_NAME and keys is None:
+            raise Exception(
+                "Can't stream invalidate all with magic current state cache"
```

Comment:
As someone who hasn't completely wrapped his head around our caching yet, what does "magic" mean in this context?

Reply:
Hopefully the bottom of https://github.com/matrix-org/synapse/blob/master/docs/tcp_replication.md#cache-invalidation-stream should explain it a bit